packages/@react-spectrum/searchfield/src/MobileSearchAutocomplete.tsx

```diff
@@ -0,0 +1,549 @@
+import {ClearButton} from '@react-spectrum/button';
+import {ComboBoxState, useComboBoxState} from '@react-stately/combobox';
+import comboboxStyles from '@react-spectrum/combobox/src/combobox.css';
+import {DismissButton} from '@react-aria/overlays';
+import {Field} from '@react-spectrum/label';
+import {FocusableRef, ValidationState} from '@react-types/shared';
+import {FocusRing, FocusScope} from '@react-aria/focus';
+import {focusSafely} from '@react-aria/focus';
+// @ts-ignore
+import intlMessages from '../intl/*.json';
+import {ListBoxBase, useListBoxLayout} from '@react-spectrum/listbox';
+import Magnifier from '@spectrum-icons/ui/Magnifier';
+import {mergeProps, useId} from '@react-aria/utils';
+import {ProgressCircle} from '@react-spectrum/progress';
+import React, {HTMLAttributes, ReactElement, ReactNode, RefObject, useCallback, useEffect, useRef, useState} from 'react';
+import {SearchFieldState, useSearchFieldState} from '@react-stately/searchfield';
+import searchStyles from '@adobe/spectrum-css-temp/components/search/vars.css';
+import {setInteractionModality, useHover} from '@react-aria/interactions';
+import {SpectrumSearchAutocompleteProps} from '@react-types/searchfield';
+import styles from '@adobe/spectrum-css-temp/components/inputgroup/vars.css';
+import {TextFieldBase} from '@react-spectrum/textfield';
+import textfieldStyles from '@adobe/spectrum-css-temp/components/textfield/vars.css';
+import {Tray} from '@react-spectrum/overlays';
+import {useButton} from '@react-aria/button';
+import {useDialog} from '@react-aria/dialog';
+import {useFilter, useMessageFormatter} from '@react-aria/i18n';
+import {useFocusableRef} from '@react-spectrum/utils';
+import {useLabel} from '@react-aria/label';
+import {useOverlayTrigger} from '@react-aria/overlays';
+import {useProviderProps} from '@react-spectrum/provider';
+import {useSearchAutocomplete} from '@react-aria/searchfield';
+
+export const MobileSearchAutocomplete = React.forwardRef(function MobileSearchAutocomplete<T extends object>(props: SpectrumSearchAutocompleteProps<T>, ref: FocusableRef<HTMLElement>) {
+  props = useProviderProps(props);
+
+  let {
+    isQuiet,
+    isDisabled,
+    validationState,
+    isReadOnly,
+    onSubmit
+  } = props;
+
+  let {contains} = useFilter({sensitivity: 'base'});
+  let comboBoxState = useComboBoxState({
+    ...props,
+    defaultFilter: contains,
+    allowsEmptyCollection: true,
+    // Needs to be false here otherwise we double up on commitSelection/commitCustomValue calls when
+    // user taps on underlay (i.e. initial tap will call setFocused(false) -> commitSelection/commitCustomValue via onBlur,
+    // then the closing of the tray will call setFocused(false) again due to cleanup effect)
+    shouldCloseOnBlur: false,
+    allowsCustomValue: true
+  });
+
+  let searchFieldState = useSearchFieldState({
+    ...props,
+    onSubmit: (value) => onSubmit(value, null)
+  });
+
+  let state = {...comboBoxState, ...searchFieldState};
```

Comment:
urgh, this feels iffy since there is a state tracker for `inputValue` and `value`. Why was this needed here but not in SearchAutocomplete?

Reply:
Good find, this should be updated to look like SearchAutocomplete. I'll fix that.